File: src/test/java/com/github/repomeisteractions/FizzBuzzTest.java
package com.github.repomeisteractions;

import org.junit.Test;

import static org.junit.Assert.*;

public class FizzBuzzTest {

    @Test
    public void isFizzNumber() {
        FizzBuzz fizzBuzz = new FizzBuzz();

        assertFalse(fizzBuzz.isFizzNumber(1));
        assertFalse(fizzBuzz.isFizzNumber(2));
        assertTrue(fizzBuzz.isFizzNumber(3));
        assertFalse(fizzBuzz.isFizzNumber(4));
        assertFalse(fizzBuzz.isFizzNumber(5));
        assertTrue(fizzBuzz.isFizzNumber(15));
    }

    @Test
    public void isBuzzNumber() {
        FizzBuzz fizzBuzz = new FizzBuzz();

        assertFalse(fizzBuzz.isBuzzNumber(1));
        assertFalse(fizzBuzz.isBuzzNumber(2));
        assertFalse(fizzBuzz.isBuzzNumber(3));
        assertFalse(fizzBuzz.isBuzzNumber(4));
        assertTrue(fizzBuzz.isBuzzNumber(5));
        assertTrue(fizzBuzz.isBuzzNumber(15));
    }

    @Test
    public void generate() {
        FizzBuzz fizzBuzz = new FizzBuzz();

        String expected = "";
        expected += "1\n";
        expected += "2\n";
        expected += "Fizz\n";
        expected += "4\n";
        expected += "Buzz\n";
        expected += "Fizz\n";
        expected += "7\n";
        expected += "8\n";
        expected += "Fizz\n";
        expected += "Buzz\n";
        expected += "11\n";
        expected += "Fizz\n";
        expected += "13\n";
        expected += "14\n";
        expected += "FizzBuzz\n";

        String result = fizzBuzz.generate(15);

        assertEquals(expected, result);
    }
}

File: fizzbuzz.test.js
const {isFizzNumber, isBuzzNumber, generate} = require("./fizzbuzz");

test('isFizzNumber', () => {
    expect(isFizzNumber(1)).toBeFalsy();
    expect(isFizzNumber(2)).toBeFalsy();
    expect(isFizzNumber(3)).toBeTruthy();
    expect(isFizzNumber(4)).toBeFalsy();
    expect(isFizzNumber(5)).toBeFalsy();
    expect(isFizzNumber(15)).toBeTruthy();
});

test('isBuzzNumber', () => {
    expect(isBuzzNumber(1)).toBeFalsy();
    expect(isBuzzNumber(2)).toBeFalsy();
    expect(isBuzzNumber(3)).toBeFalsy();
    expect(isBuzzNumber(4)).toBeFalsy();
    expect(isBuzzNumber(5)).toBeTruthy();
    expect(isBuzzNumber(15)).toBeTruthy();
});

test('generate', () => {
    let expected = "";
    expected += "1\n";
    expected += "2\n";
    expected += "Fizz\n";
    expected += "4\n";
    expected += "Buzz\n";
    expected += "Fizz\n";
    expected += "7\n";
    expected += "8\n";
    expected += "Fizz\n";
    expected += "Buzz\n";
    expected += "11\n";
    expected += "Fizz\n";
    expected += "13\n";
    expected += "14\n";
    expected += "FizzBuzz\n";

    const result = generate(15);

    expect(result).toBe(expected)
});
